tern/analyze/common.py

```diff
@@ -555,9 +556,42 @@ def get_git_sha(path_to_toplevel):
     return sha_info
 
 
+def get_git_url(path_to_toplevel):
+    '''Given a absolute path to a git repository, return the project url.'''
+    command = ['git', 'remote', '-v']
+    url_info = '(not found)'
+    try:
+        output = subprocess.check_output(  # nosec
+            command, stderr=subprocess.DEVNULL, cwd=path_to_toplevel)
+        if isinstance(output, bytes):
+            lines = output.decode('utf-8').split('\n')
+            url_dict = {}
+            for line in lines:
+                # line structure is '<label>/t<url> (fetch)'
+                split_line = line.split(' ')
+                if len(split_line) < 2:
+                    continue
+                # use fetch url
+                if split_line[1] == '(fetch)':
+                    split_line = split_line[0].split('\t')
+                    # use http or https type
+                    if split_line[1].startswith('http'):
+                        url_dict[split_line[0]] = split_line[1]
```

Comment:
You may have to do more manipulation if the remote happened to be of the form `git@`

Reply:
Also, I don't think you need to keep the remote name. You can just keep the url.